libvast/builtins/operators/summarize.cpp

```diff
@@ -729,6 +731,54 @@ class summarize_operator : public pipeline_operator {
   std::unordered_map<type, std::vector<table_slice>> blacklist_ = {};
 };
 
+/// The summarize pipeline operator implementation.
+class summarize_operator2 : public logical_operator<events, events> {
+public:
+  /// Creates a pipeline operator from its configuration.
+  /// @param config The parsed configuration of the summarize operator.
+  explicit summarize_operator2(configuration config) noexcept
+    : config_{std::move(config)} {
+    // nop
+  }
+
+  [[nodiscard]] auto
+  make_physical_operator(const type& input_schema,
+                         operator_control_plane* ctrl) noexcept
+    -> caf::expected<physical_operator<events, events>> override {
+    if (auto aggregation = aggregation::make(input_schema, config_)) {
+      return [aggregation = std::move(*aggregation), ctrl](
+               generator<table_slice> input) mutable -> generator<table_slice> {
+        for (auto&& slice : input) {
+          aggregation.add(to_record_batch(slice));
```

Comment:
```suggestion
          aggregation.add(to_record_batch(slice));
          co_yield {};
```

Reply:
What are your thoughts on this alternative?
```cpp
if (slice.rows() == 0) {
  co_yield {};
} else {
  aggregation.add(to_record_batch(slice));
}
```
I'm mainly asking because I'm not sure whether only yielding if the input actually stalled should be allowed. We might want to prohibit this, and/or document the requirements more clearly.

Reply:
They have slightly different behavior in terms of interruptibility; I think we should yield once per batch.